### PR TITLE
CEXT-3113: secrets flag implementation in create and update command

### PR DESCRIPTION
**Pull request by nvyasadobe**
## Description

This PR contains changes related to create and update cli command to handle secrets.

## Related Issue
https://jira.corp.adobe.com/browse/CEXT-3113

## Motivation and Context

This change is needed to enable secrets management in api mesh. User can provide secrets file in `--secrets` flag in create and update command.

## How Has This Been Tested?
**Verification done on local:** link api-mesh plugin on local and verify create and update command against linux and windows environment.

## Screenshots (if appropriate):
(Batch variables in windows)
image

(Invalid file extension provided)
image

(No file extension provided)
image

(Invalid argument provided)
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -391,6 +400,56 @@ async function validateAndInterpolateMesh(inputMeshData, envFilePath, command) {
...
+
+/**
+ * Read the secrets file, checks validation and interpolate mesh
+ *
+ * @param secretsFilePath Secrets file path
+ * @param command
+ */
+async function interpolateSecrets(secretsFilePath, command) {
+	try {
+		validateSecrets(secretsFilePath);
```
Comment (revanth0212) [inline src/utils.js:429]: Please move the call to the validate function outside of the interpolateSecrets function. Call the validate function from create / update functions. 

Its always better to make utility functions atomic (no side effects).

File: src/utils.js [hunk 2]
```diff
@@ -391,6 +400,56 @@ async function validateAndInterpolateMesh(inputMeshData, envFilePath, command) {
...
+		}
+		const compiledSecretsFileContent = parseEnv(secretsContent, {
+			outputFile: null,
+			options: {
+				all: false,
+				diff: false,
+				protect: false,
+				syntax: 'dollar-basic',
+			},
+			cli: false,
```
Comment (revanth0212) [inline src/utils.js:443]: Better to make this a constant and use it as here. Maybe ENV Parser config.

**Review by revanth0212 — CHANGES_REQUESTED** [review]
Nice work dude, just few suggestions.

File: src/utils.js (hunk 2 above)
Comment (nvyasadobe) [reply src/utils.js]: Done, converted to a constant

File: src/utils.js (hunk 1 above)
Comment (nvyasadobe) [reply src/utils.js]: Done, moved validate function outside.

**Pull request merged**